contracts/staking/PoSValidatorManager.sol

```diff
@@ -343,6 +346,10 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         );
         uint64 nonce = _incrementAndGetNonce(validationID);
 
+        // Check that removing this delegator would not exceed the maximum churn rate.
+        // TODO this check won't be necessary for a delegator whose validator has already initialized ending their validation period.
+        _checkAndUpdateChurnTrackerRemoval(delegator.weight);
```

Comment:
do you plan on handling this in this PR? I think we can check that the validator is PendingRemoved here, but not sure if I'm missing some other change we'd need.